### PR TITLE
[DOCS] Metricbeat: Document `unavailable` and `unknown` as `filesystem.ignore_types` values

### PR DESCRIPTION
**Pull request by jrodewig**
Documents `unavailable` and `unknown` as valid values for the `filesystem.ignore_types` array. Also adds a related example.

Closes https://github.com/elastic/beats/issues/35187.

### CODE REVIEW

File: metricbeat/module/system/filesystem/_meta/docs.asciidoc [hunk 1]
```diff
@@ -12,20 +12,26 @@ This metricset is available on:
...
 not be collected from filesystems matching these types. This setting also
 affects the `fsstats` metricset. If this option is not set, metricbeat ignores
 all types for virtual devices in systems where this information is available (e.g.
-all types marked as `nodev` in `/proc/filesystems` in Linux systems). This can be set to an empty list (`[]`)
-to make metricbeat report all filesystems, regardless of type.
+all types marked as `nodev` in `/proc/filesystems` in Linux systems).
+
+To ignore unavailable volumes, such as CD-ROM drives, include `unavailable` as a
+value in the array. To ignore unknown filesystems, include `unknown` as a value
+in the array.
```
Comment (jrodewig) [inline metricbeat/module/system/filesystem/_meta/docs.asciidoc:23]: @elastic/elastic-agent-data-plane Let me know if can elaborate more here. It seems a bit thin at the moment.

File: metricbeat/module/system/filesystem/_meta/docs.asciidoc [hunk 2]
```diff
@@ -12,20 +12,26 @@ This metricset is available on:
 [float]
 === Configuration
 
-*`filesystem.ignore_types`* - A list of filesystem types to ignore. Metrics will
+*`filesystem.ignore_types`* - An array of filesystem types to ignore. Metrics will
 not be collected from filesystems matching these types. This setting also
```
Comment (bczifra) [inline metricbeat/module/system/filesystem/_meta/docs.asciidoc:16]: >This setting also affects the `fsstats` metricset

Does it _only_ affect that metricset? The "also" could imply that it affects some un-named metricset _as well ass_ the `fsstats` metricset.

File: metricbeat/module/system/filesystem/_meta/docs.asciidoc [hunk 3]
```diff
@@ -12,20 +12,26 @@ This metricset is available on:
...
+value in the array. To ignore unknown filesystems, include `unknown` as a value
+in the array.
+
+To have metricbeat report on all filesystems, regardless of type, set
+`filesystem.ignore_types` to an empty array (`[]`).
 
 [float]
 === Filtering
 
 Often there are mounted filesystems that you do not want Metricbeat to report
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply metricbeat/module/system/filesystem/_meta/docs.asciidoc]: Good feedback. I've removed this with [`0c7880b` (#36946)](https://github.com/elastic/beats/pull/36946/commits/0c7880b04567f0bf28d37accdbd9547ecc767d38).

File: metricbeat/module/system/filesystem/_meta/docs.asciidoc (hunk 2 above)
Comment (jrodewig) [reply metricbeat/module/system/filesystem/_meta/docs.asciidoc]: Good point. I've removed the "also" and moved this sentence to the end of the paragraph. Changes in [`0c7880b` (#36946)](https://github.com/elastic/beats/pull/36946/commits/0c7880b04567f0bf28d37accdbd9547ecc767d38)

File: metricbeat/module/system/filesystem/_meta/docs.asciidoc (hunk 1 above)
Comment (jrodewig) [reply metricbeat/module/system/filesystem/_meta/docs.asciidoc]: Thanks for calling this out. I've updated the text to note that the values only applies to Windows here: https://github.com/elastic/beats/pull/36946/commits/51636bbeaf7ea6054be754fd7be005817e5d5107

Here are links to the related code:

https://github.com/elastic/gosigar/blob/master/sys/windows/syscall_windows.go#L349
https://github.com/andrewkroh/beats/blob/cbdba8aa1645a896023e387b370b5b88960f3176/metricbeat/module/system/filesystem/helper.go#L179C38-L179C46

I couldn't find the source for `unknown` but it seems to a common value based on https://github.com/elastic/beats/issues/35187

**Review by fearful-symmetry — APPROVED** [review]

**Pull request merged**